recipe-client-addon/lib/Sampling.jsm

```diff
@@ -130,4 +130,35 @@ this.Sampling = {
 
     return Sampling.isHashInBucket(inputHash, wrappedStart, end, total);
   }),
+
+  /**
+   * Sample over a list of ratios such that, over the input space, each ratio
+   * has a number of matches in correct proportion to the other ratios.
```

Comment:
I'm not actually that familiar with the statistics involved here. What I *think* this means is that if you have these ratios:
```
[1,2,3]
```
half of all input will return 2, a third will return 1, and one sixth of input will return 0. Is that right? It feels like the comment would benefit from an example (either this or some other one).